runners/generic_runner.go

```diff
@@ -0,0 +1,18 @@
+package runners
+
+import "sigs.k8s.io/controller-runtime/pkg/manager"
+
+type OperandInitializer struct {
+	cb func()
+}
+
+func NewOperandInitializer(cb func()) manager.Runnable {
+	return &OperandInitializer{
+		cb: cb,
+	}
+}
+
+func (r *OperandInitializer) Start(<-chan struct{}) error {
+	r.cb()
+	return nil
```

Comment:
instead of returning `nil`  we could then return the error from operand initialization. That also means, the OO won't start if there is a problem with the operand.

Reply:
Great idea thanks Peter. I'll implement those changes